sql/src/main/java/com/vmware/ddlog/util/sql/CalciteToH2Translator.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.ddl.SqlColumnDeclaration;
+import org.apache.calcite.sql.ddl.SqlCreateView;
+import org.apache.calcite.sql.ddl.SqlKeyConstraint;
+import org.apache.calcite.sql.parser.SqlAbstractParserImpl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.vmware.ddlog.util.sql.CalciteUtils.createCalciteParser;
+
+/**
+ * Translate some subset of SQL DDL statements from Calcite to H2.
+ */
+public class CalciteToH2Translator implements ToH2Translator<CalciteSqlStatement> {
+
+    /**
+     * Translate Calcite SQL statement to H2.
+     * @param sql
+     * @return
+     */
+    @Override
+    public H2SqlStatement toH2(CalciteSqlStatement sql) {
+        SqlAbstractParserImpl calciteParser = createCalciteParser(sql);
+        try {
+            org.apache.calcite.sql.SqlNodeList parseTree = calciteParser.parseSqlStmtList();
+            // Don't need to modify `create view`
+            if (parseTree.get(0) instanceof SqlCreateView) {
+                return new H2SqlStatement(sql.getStatement());
+            }
+            CalciteToH2 h2Translator = new CalciteToH2();
+            return new H2SqlStatement(parseTree.accept(h2Translator));
+        } catch (Exception e) {
+            throw new RuntimeException(e.getMessage());
+        }
+    }
+}
+
+/**
+ * Translates Calcite to H2 by implementing a Calcite parse node visitor.
+ * Unfortunately, Calcite and H2 aren't completely equivalent, so this method isn't just a noop.
+ * For example, H2 arrays do not accept subtypes, so we need to strip them, but this also means the Visitor needs to
+ * walk the entire parse tree and translate it into a string.
+ */
+class CalciteToH2 extends CalciteDDLVisitorBase {
+
+    @Override
+    public String visit(SqlCall call) {
+        if (call instanceof SqlColumnDeclaration) {
+            SqlColumnDeclaration castColumn = (SqlColumnDeclaration) call;
+            String type = castColumn.dataType.toString();
+
+            if (castColumn.dataType.getCollectionsTypeName() != null &&
+                    castColumn.dataType.getCollectionsTypeName().toString().equals("ARRAY")) {
+                type = "array";
+            }
+
+            String nullOperand = (castColumn.strategy == ColumnStrategy.NULLABLE) ? "null" : "not null";
+            if (castColumn.expression != null) {
+                throw new UnsupportedOperationException(
+                        "Don't know how to translate Calcite column expressions yet");
+            }
+
+            String transcribed = String.format("%s %s %s",
+                    castColumn.name.toString(),
+                    type,
+                    nullOperand);
+            translatedColumns.add(transcribed);
+            return transcribed;
+        }
+        if (call instanceof SqlKeyConstraint) {
```

Comment:
Here you can add a comment describing what happens in the implementation.
You can even give a short example input -> output.